docs/source/nlp/machine_translation.rst

```diff
@@ -0,0 +1,251 @@
+| model.{train_ds,validation_ds,test_ds}.src_file_name        | str             | null           | Path to the source language file                                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.tgt_file_name        | str             | null           | Path to the target language file                                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.tokens_in_batch      | int             | 512            | Maximum number of tokens per minibatch                                                         |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.clean                | bool            | true           | Whether to clean the dataset by discarding examples that are greater than max_seq_length       |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.max_seq_length       | int             | 512            | Path to pre-computed vocab file if exists                                                      |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.cache_ids            | bool            | true           | Whether to cache IDs to avoid re-tokenizing data                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.cache_data_per_node  | bool            | false          | Whether to cache IDs in each of the nodes in multi-node training                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.use_cache            | bool            | true           | Whether to use the cache if available                                                          |
```

Comment:
Let's mention that these are being deprecated and to use tarred dataset for caching.

Reply:
Sure, would you still suggest that I include them in this table?

Reply:
let's drop deprecated stuff from docs

Reply:
Cool, done.